test/common/alerts_firing.go

```diff
@@ -0,0 +1,66 @@
+package common
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+const deadMansSwitch = "DeadMansSwitch"
+
+func TestIntegreatlyAlertsFiring(t *testing.T, ctx *TestingContext) {
+	output, err := execToPod("curl localhost:9090/api/v1/alerts",
+		"prometheus-application-monitoring-0",
+		MonitoringOperatorNamespace,
+		"prometheus",
+		ctx)
+	if err != nil {
+		t.Fatal("Failed to exec to prometheus pod:", err)
+	}
+
+	// get all found rules from the prometheus api
+	var promApiCallOutput prometheusAPIResponse
+	err = json.Unmarshal([]byte(output), &promApiCallOutput)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal json: %s", err)
+	}
+	var alertsResult prometheusv1.AlertsResult
+	err = json.Unmarshal(promApiCallOutput.Data, &alertsResult)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal json: %s", err)
+	}
+
+	// check if any alerts other than DeadMansSwitch are firing or pending
+	var firingAlerts []string
+	var pendingAlerts []string
+	for _, alert := range alertsResult.Alerts {
+		alertName := alert.Labels["alertname"]
+
+		// dead mans switch is not firing, so fail the test
+		if alertName == deadMansSwitch && alert.State != prometheusv1.AlertStateFiring {
+			t.Fatalf("Alert: %s is not firing", deadMansSwitch)
+		}
+		// check for pending or firing alerts
+		if alertName != deadMansSwitch {
+			if alert.State == prometheusv1.AlertStateFiring {
+				firingAlerts = append(firingAlerts, string(alertName))
```

Comment:
Hi, my suggestion is to log immediately the errors using `t.Errorf` and then continue to the next alert instead of pushing the errors in an array. In this way, you will not need all the additionals `if` at the end of the test

Example:
https://github.com/integr8ly/integreatly-operator/blob/69161d395707aefabacf7dd12e1e851d672d61be/test/common/deployment_types.go#L203

Reply:
Cool will do, thanks! 
Do I still need the `t.Fatal()` at the end then? Or will the test fail regardless if there's a `t.Errorf()`?

Reply:
No there is no need for the `t.Fatal()` anymore, the test will fail if just one of the `t.Errorf` is called

Reply:
Cool, thanks Davide!